src/js/MainController.js

```diff
@@ -140,7 +140,9 @@ goog.require('ga_storage_service');
               if (!$window.olcs) {
                 // lazy load cesium library
                 $.getScript(ol3CesiumLibUrl, function() {
-                  $scope.ol3d = loadCesiumViewer($scope.map, active);
+                  $scope.$applyAsync(function() {
```

Comment:
I think I see what you mean. You're trying to piggy-backing on an already-scheduled digest cycle.